### PR TITLE
newSpecFile should update scan_id if provided

### LINKED ISSUE #271: BUG: newSpecFile() does not reset the scan ID
As Zhan Zhang (APS) reported:

> [`newSpecFile('zztestbp', scan_id=1)`](https://github.com/BCDA-APS/bluesky_training/blob/153352e5b91c82cb0869891df89081bde1141a35/bluesky/instrument/callbacks/spec_data_file_writer.py#L51)
> Is that the right one to use?  I don't recall all the details, but I think the scan_id=1 was ignored when calling [`specwriter.newfile(fname, **kwargs)`](https://github.com/BCDA-APS/apstools/blob/0d7647f66376336186c0f2cf3a7b88730ad6b5b9/apstools/callbacks/spec_file_writer.py#L603C52-L603C54)

### PR DESCRIPTION
**Pull request by prjemian**
- close #271

### CODE REVIEW

**Comment by prjemian** [comment]
Had to make a function that searched for the `RE` object.  The search walks backwards through the call stack to find the closest definition.  The user can always provide the `RE=RE` kwarg to be more specific.

File: bluesky/instrument/callbacks/spec_data_file_writer.py [hunk 1]
```diff
@@ -48,6 +48,32 @@ def spec_comment(comment, doc=None):
...
+    * Search backwards through each frame in the stack.
+        * Find any RunEngine objects.
+            * Prefer one with name 'RE'.
+            * Otherwise, pick the first one.
+    * Otherwise, return 'None'.
+    
+    suggestion: Hoist to apstools.utils
+    """
+    import inspect
+    from bluesky import RunEngine
```
Comment (MDecarabas) [inline bluesky/instrument/callbacks/spec_data_file_writer.py:64]: Same as in previous PR. Let's try to make these dependencies global

File: bluesky/tests/test_newSpecFile.py [hunk 2]
```diff
@@ -61,6 +64,37 @@ def test_newSpecFile(title, scan_id, tempdir):
...
+
+    # repeat, without passing RE as parameter
+    RE.md["scan_id"] = INITIAL_SCAN_ID
+    assert RE.md["scan_id"] == INITIAL_SCAN_ID
+
+    newSpecFile("title", scan_id=scan_id)
+    assert RE.md["scan_id"] != INITIAL_SCAN_ID
+    assert RE.md["scan_id"] == scan_id
+
+
```
Comment (MDecarabas) [inline bluesky/tests/test_newSpecFile.py:97]: Good testing!

File: bluesky/instrument/callbacks/spec_data_file_writer.py (hunk 1 above)
Comment (prjemian) [reply bluesky/instrument/callbacks/spec_data_file_writer.py:64]: I prefer to encapsulate both these imports to this one function.  If other functions eventually need either, then it is reasonable to move the import to module scope.  It is not necessary at this time to move them to module level.

Comment (MDecarabas) [reply bluesky/instrument/callbacks/spec_data_file_writer.py:64]: sounds good

**Review by MDecarabas — APPROVED** [review]
Solve conflicts in .rst file and we should be ok

**Pull request merged**